[BUG] test_round_robin_sort_fallback failed with ValueError: 'a_1' is not in list

In the nightly Dataproc integration tests for 21.08, test_round_robin_sort_fallback failed:
```
[2021-07-19T12:07:52.837Z] =================================== FAILURES ===================================
[2021-07-19T12:07:52.837Z] [31m[1m_ test_round_robin_sort_fallback[[('a', Struct(('a_1', Struct(('a_1_1', Integer),('a_1_2', Float),('a_1_3', Double))),('b_1', Long)))]] _[0m
[2021-07-19T12:07:52.837Z] 
[2021-07-19T12:07:52.837Z] self = Row(a=None, extra=1), item = 'a_1'
[2021-07-19T12:07:52.837Z] 
[2021-07-19T12:07:52.837Z]     [94mdef[39;49;00m [92m__getitem__[39;49;00m([96mself[39;49;00m, item):
[2021-07-19T12:07:52.837Z]         [94mif[39;49;00m [96misinstance[39;49;00m(item, ([96mint[39;49;00m, [96mslice[39;49;00m)):
[2021-07-19T12:07:52.837Z]             [94mreturn[39;49;00m [96msuper[39;49;00m(Row, [96mself[39;49;00m).[92m__getitem__[39;49;00m(item)
[2021-07-19T12:07:52.837Z]         [94mtry[39;49;00m:
[2021-07-19T12:07:52.837Z]             [90m# it will be slow when it has many fields,[39;49;00m
[2021-07-19T12:07:52.837Z]             [90m# but this will not be used in normal cases[39;49;00m
[2021-07-19T12:07:52.837Z] >           idx = [96mself[39;49;00m.__fields__.index(item)
[2021-07-19T12:07:52.837Z] [1m[31mE           ValueError: 'a_1' is not in list[0m
[2021-07-19T12:07:52.837Z] 
[2021-07-19T12:07:52.837Z] [1m[31m/hadoop/yarn/nm-local-dir/usercache/root/appcache/application_1626687042683_0001/container_e01_1626687042683_0001_01_000001/pyspark.zip/pyspark/sql/types.py[0m:1558: ValueError
[2021-07-19T12:07:52.837Z] 
[2021-07-19T12:07:52.837Z] [33mDuring handling of the above exception, another exception occurred:[0m
[2021-07-19T12:07:52.837Z] 
[2021-07-19T12:07:52.837Z] data_gen = [('a', Struct(('a_1', Struct(('a_1_1', Integer),('a_1_2', Float),('a_1_3', Double))),('b_1', Long)))]
[2021-07-19T12:07:52.837Z] 
[2021-07-19T12:07:52.837Z]     [37m@allow_non_gpu[39;49;00m([33m'[39;49;00m[33mShuffleExchangeExec[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mRoundRobinPartitioning[39;49;00m[33m'[39;49;00m)
[2021-07-19T12:07:52.837Z]     [37m@pytest[39;49;00m.mark.parametrize([33m'[39;49;00m[33mdata_gen[39;49;00m[33m'[39;49;00m, [[([33m'[39;49;00m[33mag[39;49;00m[33m'[39;49;00m, ArrayGen(string_gen))],
[2021-07-19T12:07:52.837Z]         [([33m'[39;49;00m[33ma[39;49;00m[33m'[39;49;00m, StructGen([
[2021-07-19T12:07:52.837Z]           ([33m'[39;49;00m[33ma_1[39;49;00m[33m'[39;49;00m, StructGen([
[2021-07-19T12:07:52.837Z]             ([33m'[39;49;00m[33ma_1_1[39;49;00m[33m'[39;49;00m, int_gen),
[2021-07-19T12:07:52.837Z]             ([33m'[39;49;00m[33ma_1_2[39;49;00m[33m'[39;49;00m, float_gen),
[2021-07-19T12:07:52.837Z]             ([33m'[39;49;00m[33ma_1_3[39;49;00m[33m'[39;49;00m, double_gen)
[2021-07-19T12:07:52.837Z]           ])),
[2021-07-19T12:07:52.837Z]           ([33m'[39;49;00m[33mb_1[39;49;00m[33m'[39;49;00m, long_gen)
[2021-07-19T12:07:52.837Z]         ]))],
[2021-07-19T12:07:52.837Z]         [([33m'[39;49;00m[33ma[39;49;00m[33m'[39;49;00m, simple_string_to_string_map_gen)]], ids=idfn)
[2021-07-19T12:07:52.837Z]     [37m@ignore_order[39;49;00m(local=[94mTrue[39;49;00m) [90m# To avoid extra data shuffle by 'sort on Spark' for this repartition test.[39;49;00m
[2021-07-19T12:07:52.837Z]     [94mdef[39;49;00m [92mtest_round_robin_sort_fallback[39;49;00m(data_gen):
[2021-07-19T12:07:52.837Z]         [94mfrom[39;49;00m [04m[96mpyspark[39;49;00m[04m[96m.[39;49;00m[04m[96msql[39;49;00m[04m[96m.[39;49;00m[04m[96mfunctions[39;49;00m [94mimport[39;49;00m lit
[2021-07-19T12:07:52.837Z] >       assert_gpu_fallback_collect(
[2021-07-19T12:07:52.837Z]                 [90m# Add a computed column to avoid shuffle being optimized back to a CPU shuffle like in test_repartition_df[39;49;00m
[2021-07-19T12:07:52.837Z]                 [94mlambda[39;49;00m spark : gen_df(spark, data_gen).withColumn([33m'[39;49;00m[33mextra[39;49;00m[33m'[39;49;00m, lit([94m1[39;49;00m)).repartition([94m13[39;49;00m),
[2021-07-19T12:07:52.837Z]                 [33m'[39;49;00m[33mShuffleExchangeExec[39;49;00m[33m'[39;49;00m)
[2021-07-19T12:07:52.837Z] 
[2021-07-19T12:07:52.837Z] [1m[31mintegration_tests/src/main/python/repart_test.py[0m:141: 
[2021-07-19T12:07:52.837Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2021-07-19T12:07:52.837Z] [1m[31mintegration_tests/src/main/python/asserts.py[0m:326: in assert_gpu_fallback_collect
[2021-07-19T12:07:52.837Z]     assert_equal(from_cpu, from_gpu)
[2021-07-19T12:07:52.837Z] [1m[31mintegration_tests/src/main/python/asserts.py[0m:94: in assert_equal
[2021-07-19T12:07:52.837Z]     _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
[2021-07-19T12:07:52.837Z] [1m[31mintegration_tests/src/main/python/asserts.py[0m:41: in _assert_equal
[2021-07-19T12:07:52.837Z]     _assert_equal(cpu[index], gpu[index], float_check, path + [index])
[2021-07-19T12:07:52.837Z] [1m[31mintegration_tests/src/main/python/asserts.py[0m:34: in _assert_equal
[2021-07-19T12:07:52.837Z]     _assert_equal(cpu[field], gpu[field], float_check, path + [field])
[2021-07-19T12:07:52.837Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2021-07-19T12:07:52.837Z] 
[2021-07-19T12:07:52.837Z] self = Row(a=None, extra=1), item = 'a_1'
[2021-07-19T12:07:52.837Z] 
[2021-07-19T12:07:52.837Z]     [94mdef[39;49;00m [92m__getitem__[39;49;00m([96mself[39;49;00m, item):
[2021-07-19T12:07:52.837Z]         [94mif[39;49;00m [96misinstance[39;49;00m(item, ([96mint[39;49;00m, [96mslice[39;49;00m)):
[2021-07-19T12:07:52.837Z]             [94mreturn[39;49;00m [96msuper[39;49;00m(Row, [96mself[39;49;00m).[92m__getitem__[39;49;00m(item)
[2021-07-19T12:07:52.837Z]         [94mtry[39;49;00m:
[2021-07-19T12:07:52.837Z]             [90m# it will be slow when it has many fields,[39;49;00m
[2021-07-19T12:07:52.837Z]             [90m# but this will not be used in normal cases[39;49;00m
[2021-07-19T12:07:52.837Z]             idx = [96mself[39;49;00m.__fields__.index(item)
[2021-07-19T12:07:52.837Z]             [94mreturn[39;49;00m [96msuper[39;49;00m(Row, [96mself[39;49;00m).[92m__getitem__[39;49;00m(idx)
[2021-07-19T12:07:52.837Z]         [94mexcept[39;49;00m [96mIndexError[39;49;00m:
[2021-07-19T12:07:52.837Z]             [94mraise[39;49;00m [96mKeyError[39;49;00m(item)
[2021-07-19T12:07:52.837Z]         [94mexcept[39;49;00m [96mValueError[39;49;00m:
[2021-07-19T12:07:52.837Z] >           [94mraise[39;49;00m [96mValueError[39;49;00m(item)
[2021-07-19T12:07:52.837Z] [1m[31mE           ValueError: a_1[0m
```

Looks similar to #2572.

--- a/integration_tests/src/main/python/repart_test.py
+++ b/integration_tests/src/main/python/repart_test.py
@@ -132,7 +132,8 @@ def test_repartition_df(data_gen, num_parts, length):
         ('a_1_2', float_gen),
         ('a_1_3', double_gen)
       ])),
-      ('b_1', long_gen)
+      ('b_1', long_gen),
+      ('c_1', int_gen)
     ]))],
     [('a', simple_string_to_string_map_gen)]], ids=idfn)
 @ignore_order(local=True) # To avoid extra data shuffle by 'sort on Spark' for this repartition test.

